src/WebJobs.Script.Grpc/Channel/GrpcCapabilities.cs

```diff
@@ -26,19 +26,36 @@ public string GetCapabilityState(string capability)
             return null;
         }
 
-        public void UpdateCapabilities(IDictionary<string, string> capabilities)
+        public void UpdateCapabilities(IDictionary<string, string> capabilities, GrpcCapabilitiesUpdateStrategy strategy)
         {
             if (capabilities == null)
             {
                 return;
             }
 
-            _logger.LogDebug($"Updating capabilities: {capabilities.ToString()}");
+            _logger.LogDebug($"Updating capabilities using {strategy} strategy. Current: {_capabilities} Incoming: {capabilities}");
```

Comment:
This log message will not print the content of the dictionary. It will print something like
 `Current: System.Collections.Generic.Dictionary2[System.String,System.String]`

Reply:
We should also use the following logging pattern:

```suggestion
            _logger.LogDebug("Updating capabilities using {strategy} strategy. Current: {_capabilities} Incoming: {capabilities}", strategy, current, incoming);
```